### PR TITLE
Upgrade Scala from 2.10 to 2.12

### PR DESCRIPTION
**Pull request by adityagupta94**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -267,6 +268,20 @@
...
+                <executions>
+                    <execution>
+                        <id>rat-check</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>check</goal>
+                        </goals>
+                        <configuration>
+                            <excludes>
+                                <exclude>*.md</exclude>
```
Comment (tivv) [inline pom.xml]: why? Just add a license in HTML comment <-- -->

File: pom.xml [hunk 2]
```diff
@@ -696,9 +711,14 @@
                 <artifactId>asm-commons</artifactId>
                 <version>${asm.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.scala-lang</groupId>
+                <artifactId>scala-library</artifactId>
+                <version>${scala.version}</version>
```
Comment (tivv) [inline pom.xml]: Why do you need scala as a compile dependency?

Comment (tivv) [reply pom.xml]: If it's to override version, it's better to use dependency management

File: pom.xml (hunk 1 above)
Comment (dli357) [reply pom.xml]: Discussed with @tivv offline, this should be ok as we should be excluding markdown files.

File: pom.xml (hunk 2 above)
Comment (dli357) [reply pom.xml]: Does dependencyManagement override transitive dependencies?

**Review by dli357 — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (tivv) [reply pom.xml]: Yes, that's the best part about it!

Comment (adityagupta94) [reply pom.xml]: Yeah, made this change and tested it.

**Pull request merged**